src/test/android/testapp/src/androidTest/java/software/amazon/awssdk/crt/test/android/CrtPlatformImpl.java

```diff
@@ -0,0 +1,136 @@
+        Bundle testArgs = InstrumentationRegistry.getArguments();
+        if (testArgs != null) {
+            final Set<String> keys = testArgs.keySet();
+            for (final String key : keys) {
+                if (key.startsWith("aws.crt.")) {
+                    System.setProperty(key, testArgs.getString(key));
+                }
+            }
+        }
+    }
+
+    public PackageInfo.Version getVersion() {
+        return new PackageInfo.Version("0.0.0-UNITTEST");
+    }
+
+    // Used to avoid exception when the property being copied is null (not set)
+    private void SetPropertyFromProperty(String propertyName, String propertyToCopy){
+        if (System.getProperty(propertyToCopy) != null) {
+            System.setProperty(propertyName, System.getProperty(propertyToCopy));
+        } else {
+            System.out.println("Android TEST: CrtPlatformImpl.SetPropertyFromProperty() Property:" + propertyToCopy + " is null");
+        }
+    }
+
+    // Set a System property to the contents of a file in the assets folder
+    private void SetPropertyFromFile(String propertyName, String fileName){
+        AssetManager assets = InstrumentationRegistry.getInstrumentation().getTargetContext().getResources().getAssets();
+        try (InputStream assetStream = assets.open(fileName);) {
+            byte[] contents = new byte[assetStream.available()];
+            assetStream.read(contents);
+            System.setProperty(propertyName, new String(contents).trim());
+        } catch (IOException ex){
+            System.out.println("Android TEST: CrtPlatformImpl.SetPropertyFromFile(" + propertyName + ", " + fileName + ") IOException: " + ex.toString());
+        }
+    }
+
+    // Attempts to create a cached file from a file in the assets folder and sets a System property pointing to the created file
+    // CRT cannot directly access files in the assets folder of an Android app as they are compressed
+    private void SetPropertyToFileLocation(String propertyName, String fileName){
+        AssetManager assets = InstrumentationRegistry.getInstrumentation().getTargetContext().getResources().getAssets();
+        String cachedName = InstrumentationRegistry.getInstrumentation().getTargetContext().getExternalCacheDir().getAbsolutePath() + "/" + fileName;
+
+        try (InputStream assetStream = assets.open(fileName);
+            FileOutputStream cachedRes = new FileOutputStream(cachedName)){
+                byte[] data = new byte[assetStream.available()];
+                assetStream.read(data);
+                cachedRes.write(data);
+                System.setProperty(propertyName, cachedName);
+            } catch (IOException ex) {
+                System.out.println("Android TEST: CrtPlatformImpl.SetPropertyToFileLocation(" + propertyName + ", " + fileName + ") IOException: " + ex.toString());
+            }
+    }
+
+    public void testSetup(Object context) {
```

Comment:
With the Java Junit test, I found the @BeforeClass flag which allows you to declare a setup function that is called only once for all tests, and removed `are.test.properties.setup`. [setupOnce()](https://github.com/awslabs/aws-crt-java/blob/1fc22a8d3a05ca27616e3878affc6b34eed6c2cd/src/test/java/software/amazon/awssdk/crt/test/CrtTestFixture.java#L210)

It would be nice if we can make relative changes so that `CrtPlatformImpl` differentiate the setup that only called once and the setup called for each test.